File: python-version/LC1304/sum_zero.py
from typing import List


class Solution:
    @staticmethod
    def sumZero(n: int) -> List[int]:
        ans = [0] * n
        low, high = 0, n - 1
        k = n
        while low < high:
            ans[low], ans[high] = -k, k
            low += 1
            high -= 1
            k -= 1
        if n & 1 == 1:
            ans[low] = 0
        return ans

File: go-version/leetcode/LC1304/sum_zero.go
package LC1304

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/9/7 下午10:22
 * @FilePath: sum_zero
 * @Description: 按照相反数依次填入即可
 */

func sumZero(n int) []int {
	ans := make([]int, n)
	low, high := 0, n-1
	k := n
	for low < high {
		ans[low] = -k
		ans[high] = k
		k--
		low++
		high--
	}
	if n&1 == 1 {
		ans[low] = 0
	}
	return ans
}
